### PR TITLE
feat: Fir 38088 allow passing custom headers in the go sdk

### PR DESCRIPTION
**Pull request by stepansergeevitch**
Added ability to pass additional headers as a map in a context `additionalHeaders` variable

### CODE REVIEW

File: client_base.go [hunk 1]
```diff
@@ -228,6 +228,23 @@ type requestParameters struct {
...
 
+type ContextKey string
+
+func extractAdditionalHeaders(ctx context.Context) map[string]string {
+	additionalHeaders, ok := ctx.Value(ContextKey("additionalHeaders")).(map[string]string)
+	if ok {
+		// only take headers that start with Firebolt- prefix
+		filteredHeaders := make(map[string]string)
+		for key, value := range additionalHeaders {
+			if strings.HasPrefix(key, "Firebolt-") {
```
Comment (alexkaplun-firebolt) [inline client_base.go:239]: Have we considered using `X-Firebolt-**` pattern?  `X` is a standard prefix for HTTP extension headers, and we are likely to end up with more custom headets set (see https://docs.google.com/document/d/1KqjIOrDLW4O1Y34pLr5oGa7f20tgc4Cx9Ae84oOMqQM/edit?tab=t.0#heading=h.ev90bjt1onfz)

Comment (stepansergeevitch) [reply client_base.go:239]: Oh, ok, I can update it then.

Comment (alexkaplun-firebolt) [reply client_base.go]: @cosmpop FYI

Comment (alexkaplun-firebolt) [reply client_base.go]: I'm not sure if this would require any update on packdb

**Pull request merged**